alacritty_terminal/src/ansi.rs

```diff
@@ -25,60 +25,88 @@ use vte;
 use crate::term::color::Rgb;
 
 // Parse color arguments
-//
-// Expect that color argument looks like "rgb:xx/xx/xx" or "#xxxxxx"
-fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
-    let mut iter = color.iter();
 
-    macro_rules! next {
-        () => {
-            iter.next().map(|v| *v as char)
-        };
+// Parse a color argument in the form "#xxxxxx"
+// The use of this format is no longer encouraged by xparsecolor
+fn parse_legacy_rgb_color(color: &[u8]) -> Option<Rgb> {
+    if color.len() < 7 || color[0] != b'#' {
+        return None;
     }
 
+    let mut iter = color.iter().map(|v| (*v as char));
+    iter.next();
```

Comment:
Instead of creating an iterator and then calling `iter.next()`, it should be possible to just add `.skip(1)` after the `map`.